client/cron/CdApplicationStatusUpdateHandler.go

```diff
@@ -141,6 +157,8 @@ func (impl *CdApplicationStatusUpdateHandlerImpl) Subscribe() error {
 }
 
 func (impl *CdApplicationStatusUpdateHandlerImpl) HelmApplicationStatusUpdate() {
+	cronProcessStartTime := time.Now()
+	defer middleware.DeploymentStatusCronDuration.WithLabelValues(pipeline.DEVTRON_APP_HELM_PIPELINE_STATUS_UPDATE_CRON).Observe(time.Since(cronProcessStartTime).Seconds())
```

Comment:
i think this would not work properly, u need to pass time.Since in a block, so that it calculates value at defer call time.

Reply:
updated